### PR TITLE
WINDUP-686  Windup Should Automatically Cleanup Graph & Archive D…

### PR DESCRIPTION
**Pull request by OndraZizka**
…irectories WIP - NPE

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/464/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1095/<h2>Failed Tests: <span class='status-failure'>7</span></h2><h3><a name='org.jboss.windup.ui:windup-ui-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport'>org.jboss.windup.ui:windup-ui-tests</a>: <span class='status-failure'>7</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testOverwriteConfirmation/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testOverwriteConfirmation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testUserRulesDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testUserRulesDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testDuplicateUserRulesDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testDuplicateUserRulesDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testNewMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testNewMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testUserIgnoreDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testUserIgnoreDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testOutputDirCannotBeParentOfInputDir/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testOutputDirCannotBeParentOfInputDir</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1095/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testOutputDefaultValue/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testOutputDefaultValue</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/465/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1096/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1099/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/469/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1100/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/470/
Test PASSed.

File: config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderRegistryCache.java [hunk 1]
```diff
@@ -111,7 +112,8 @@ public RuleProviderRegistry getRuleProviderRegistry()
             }
             catch (Exception e)
             {
-                LOG.log(Level.WARNING, "Failed to load rule information due to: " + e.getMessage(), e);
+                ///LOG.log(Level.WARNING, "Failed to load rule information due to: " + e.getMessage(), e);
+                throw new WindupException("Failed to load rule information due to: " + e.getMessage(), e);
```
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/metadata/RuleProviderRegistryCache.java]: Needs a cleanup

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1101/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1105/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/474/
Test PASSed.

**Pull request merged**